### ISSUE TITLE
Allow for Switching between videos in a scene. maybe a scene=3?

### ISSUE THREAD

**Issue by steveseguin** [steveseguin/obsninja#550]
I played around a bit with the director control room. And I would want to use the Add To Scene feature to push the seleceted camera/screen to the empty Scene Link. Works good of course. But then I would like to have a feature where I can select another screen and use Add To Scene to OVERTAKE the previous camera/screen. Right now I can only make it that I either have to close down the first camera and then add the second camera, which will result in an empty screen for a mili second to a second. Or I could add the second together with the first camera, resulting in them being side by side or one on top of the other. I would like to have a seamless switch between the cameras.  -- EcksDee on discord

**Comment by eazuka** [comment]
This will be really awesome, i am actually searching for the possibility of this being a hidden feature which lead me to this wiki page and the i saw this. This is highly needed.

**Comment by steveseguin** [comment]
Created a scene for this action.  It's crude, but it will show only the last video added to a scene.  
```https://obs.ninja/alpha/?room=XXX&scene=2```

**Comment by chrismarquardt** [comment]
Awesome! 

Notes: 

1. the unused "add to scene" buttons in the control center should unlatch when another one is pressed
2. make the "add to scene" buttons accessible through simple keyboard shortcuts (1 2 3 4 ...)

**Comment by chrismarquardt** [comment]
Interesting behavior: 

I open https://obs.ninja/alpha/?r=x&scene=2 in a chrome tab, I don't click it (re: autostart behavior)

I then "add to scene" from director and the scene=2 tab only shows the first frame from the video of each guest I "add to scene" - only when I click in the window will the video start streaming.

So it seems to be in some limbo state between streaming and not streaming.

**Comment by steveseguin** [comment]
I think this is caused by me tying to play the audio after the video track has started.  Since Chrome requires a user-gesture to a video play with audio, there is a brief moment where the video is muted, gets displayed, but then is stopped by Chrome when I try to play the audio.  Clicking the tab allows it to play as intended.

**Comment by chrismarquardt** [comment]
Is there any way for you to detect if the tab is playing or not? If it's not playing you could display a "start streaming" button or link

**Comment by steveseguin** [comment]
Yes.  I auto-play on any user input and continue to check every few seconds to see if I did get permissions to play also.

I had at some point thought of adding a big play Icon overlay the video if it stopped;  for scenes though, I don't want to add any obtuse UI or graphics that are not 'clean' -- I didn't want to surprise a producer with some play button accidentally.    I'm open to reconsidering however.

That said, this is a non-issue in OBS or the Electron app -- both don't have the user-gesture issue and scenes don't really get used in Chrome --

**Comment by chrismarquardt** [comment]
I think a button would be okay if it was for the first "it's not playing" event and not for subsequent ones, as it tends to keep playing once it started. This way it wouldn't be a surprise and help with discoverability and reduce confusion. And could be hidden through &clean flag

**Comment by chrismarquardt** [comment]
(main reason I'm asking btw. is the simple video conference use case with my family. "yes, mom, now click into the browser")

**Comment by steveseguin** [comment]
is this as a scene or a guest?

**Comment by chrismarquardt** [comment]
it was actually a scene that I wanted someone else to screen grab, he sat in front of a black browser for a minute until I realized he didn't click

**Comment by chrismarquardt** [comment]
guests will be fine, they clicked anyway

**Comment by steveseguin** [comment]
I'll give it a go.  :)

**Comment by steveseguin** [comment]
![image](https://user-images.githubusercontent.com/2575698/100906287-0562d000-3497-11eb-8564-8374929450c0.png)

**Comment by chrismarquardt** [comment]
![D6BEC708-984B-424D-865C-13BFF4097681](https://user-images.githubusercontent.com/2527608/100907583-c9496280-34ca-11eb-9a1e-e63aaba8ad80.jpeg)

Maybe a smidgeon smaller, so it’s more in line with other player a out there. Transfer of learning, people will identify it easier as something to click that way.

**Comment by chrismarquardt** [comment]
(That’s just me being nitpicky 😁)

**Comment by steveseguin** [comment]
![image](https://user-images.githubusercontent.com/2575698/100908956-195c0100-349a-11eb-9bb3-88eb73972960.png)

**Comment by chrismarquardt** [comment]
<3

**Comment by chrismarquardt** [comment]
close?

**Comment by steveseguin** [comment]
There's a lingering issue where the "add to scene" button doesn't toggle off the other scene buttons.  I need to create a dedicated button for scene=2 I think to complete this task.

**Comment by nicknurock** [comment]
When using scene=2 on production, the swapping works correctly, other than the lingering toggle issue you mention.  However if the Director chooses to enable their microphone, the swapping seems to stop working.  The scene goes blank on the next click of "add to scene" and no further guest video will appear.  The only way to get scene2 working again seems to be for the Director to reload their page.  

Amazing software, Steve, thank you.

**Comment by steveseguin** [comment]
Closing old issue automatically by bot; manual scenes and add-to-scene controls now cover this.
